.github/workflows/publish_pytorch_dev_docker.yml

```diff
@@ -72,3 +72,66 @@ jobs:
           push: true
           tags: ${{ steps.meta.outputs.tags }}
           labels: ${{ steps.meta.outputs.labels }}
+  verify-image:
+    needs: build-and-push-image
+    strategy:
+      matrix:
+        targets:
+          - "gfx942"
+    uses: ./.github/workflows/verify_docker_image.yml
+    with:
+      amdgpu_target: ${{ matrix.targets }}
+  on-success:
+    needs: verify-image
+    if: ${{ needs.verify-image.result == 'success' }}
+    strategy:
+      matrix:
+        targets:
+          - "gfx942"
+    runs-on: azure-linux-scale-rocm
+    steps:
+      - name: Log in to the Container registry
+        uses: docker/login-action@9780b0c442fbb1117ed29e0efdff1e18412f7567 # v3.3.0
+        with:
+          registry: ghcr.io
+          username: ${{ github.actor }}
+          password: ${{ secrets.GITHUB_TOKEN }}
+      - name: Write pinned version to file
+        run: |
+          echo "${{ env.VERSION }}" > latest_verified_image_version_pinned.txt
+      - name: Upload pinned version file as an artifact
+        uses: actions/upload-artifact@4cec3d8aa04e39d1a68397de0c4cd6fb9dce8ec1 # v4.6.1
+        with:
+          name: latest_verified_image_version_pinned-${{ matrix.targets }}
+          path: latest_verified_image_version_pinned.txt
+      - name: Tag and Push 'verified'
+        run: |
+          IMAGE=ghcr.io/rocm/therock_pytorch_dev_ubuntu_24_04_${{ matrix.targets }}
+          docker pull $IMAGE:main
+          docker tag $IMAGE:main $IMAGE:verified
+          docker push $IMAGE:verified
+      - name: Print success info
+        run: echo "Verification passed and tagged as 'verified'."
+  on-failure:
+    needs: verify-image
+    if: ${{ needs.verify-image.result == 'failure' }}
+    strategy:
+      matrix:
+        targets:
+          - "gfx942"
+    runs-on: azure-linux-scale-rocm
+    steps:
+      - name: Log in to the Container registry
+        uses: docker/login-action@9780b0c442fbb1117ed29e0efdff1e18412f7567 # v3.3.0
+        with:
+          registry: ghcr.io
+          username: ${{ github.actor }}
+          password: ${{ secrets.GITHUB_TOKEN }}
+      - name: Tag and Push 'verification-failed'
```

Comment:
Do we want a 'verification-failed' tag at all? If there is no tag verification obviously did not pass.